Sources/DistributedSystem/DistributedSystemServer.swift

```diff
@@ -137,20 +137,29 @@ public class DistributedSystemServer: DistributedSystem {
         metadata[ServiceMetadata.systemName.rawValue] = systemName
         metadata[ServiceMetadata.processIdentifier.rawValue] = String(ProcessInfo.processInfo.processIdentifier)
         metadata[ServiceMetadata.moduleIdentifier.rawValue] = String(moduleID.rawValue)
+
         let (serviceID, updateHealthStatus) = super.addService(name, metadata, factory)
         let future = registerService(name, serviceID, metadata: metadata)
+        var registrationError: Error?
         do {
             try await future.get()
         } catch {
-            logger.error("future.get() failed: \(error)")
-            throw error
+            registrationError = error
```

Comment:
Can we instead re-order health status update scheduling and checking future status?

Reply:
Fixed.